sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -105,3 +113,33 @@ def set_cache_manager(self, cache_manager):
   def cache_manager(self):
     """Gets the cache manager held by current Interactive Environment."""
     return self._cache_manager
+
+  def set_pipeline_result(self, pipeline, result):
```

Comment:
Why do we need this method? It looks like it's only being used in tests.

Reply:
It's not used yet. It should be invoked by the InteractiveRunner once we have all the building blocks ready.
What we are planning to do is to only allow one job for one user pipeline running at the same time within a runner instance.
So if the user re-executes p.run() for an async running job.  The runner should cancel the running job using current tracked pipeline_result and starts a new one, then track the new  pipeline_result.